### ISSUE TITLE
Improve stdlib module initialization error handling.

### ISSUE THREAD

**Issue by brandtbucher** [python/cpython#83004]
BPO | [38823](https://bugs.python.org/issue38823)
--- | :---
Nosy | @Yhg1s, @vstinner, @asvetlov, @serhiy-storchaka, @miss-islington, @brandtbucher, @shihai1991, @erlend-aasland
PRs | <li>python/cpython#17195</li><li>python/cpython#17196</li><li>python/cpython#17197</li><li>python/cpython#17198</li><li>python/cpython#17199</li><li>python/cpython#17200</li><li>python/cpython#17206</li><li>python/cpython#17215</li><li>python/cpython#17216</li><li>python/cpython#17226</li><li>python/cpython#17227</li><li>python/cpython#17235</li><li>python/cpython#17236</li><li>python/cpython#17250</li><li>python/cpython#17260</li><li>python/cpython#17263</li><li>python/cpython#17264</li><li>python/cpython#17271</li><li>python/cpython#17272</li><li>python/cpython#17276</li><li>python/cpython#17280</li><li>python/cpython#17281</li><li>python/cpython#17282</li><li>python/cpython#17283</li><li>python/cpython#17298</li><li>python/cpython#18365</li><li>python/cpython#23247</li><li>python/cpython#23248</li><li>python/cpython#23258</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/brandtbucher'
closed_at = None
created_at = <Date 2019-11-16.18:54:25.878>
labels = ['3.7', '3.8', 'type-bug', 'library', '3.9']
title = 'Improve stdlib module initialization error handling.'
updated_at = <Date 2020-11-13.13:44:17.121>
user = 'https://github.com/brandtbucher'
```

bugs.python.org fields:
```python
activity = <Date 2020-11-13.13:44:17.121>
actor = 'vstinner'
assignee = 'brandtbucher'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2019-11-16.18:54:25.878>
creator = 'brandtbucher'
dependencies = []
files = []
hgrepos = []
issue_num = 38823
keywords = ['patch']
message_count = 30.0
messages = ['356764', '356765', '356773', '356774', '356775', '356781', '356782', '356783', '356809', '356820', '356872', '356873', '356874', '356984', '357002', '357003', '357004', '357014', '357019', '357029', '357044', '357046', '357047', '357050', '357051', '357114', '361486', '380822', '380824', '380883']
nosy_count = 8.0
nosy_names = ['twouters', 'vstinner', 'asvetlov', 'serhiy.storchaka', 'miss-islington', 'brandtbucher', 'shihai1991', 'erlendaasland']
pr_nums = ['17195', '17196', '17197', '17198', '17199', '17200', '17206', '17215', '17216', '17226', '17227', '17235', '17236', '17250', '17260', '17263', '17264', '17271', '17272', '17276', '17280', '17281', '17282', '17283', '17298', '18365', '23247', '23248', '23258']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue38823'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9']
```

</p></details>

* PR: gh-98840

* PR: gh-98841

* PR: gh-98842

* PR: gh-99043

* PR: gh-99044

* PR: gh-99059

* PR: gh-99060

* PR: gh-99163

* PR: gh-99164

* PR: gh-99165

* PR: gh-99166

* PR: gh-103261

* PR: gh-103383

* PR: gh-103385

* PR: gh-103386

* PR: gh-103420

**Comment by brandtbucher** [comment]
I know I'm a bit late, but thanks so much for picking up this old project @hauntsaninja!

**Comment by erlend-aasland** [comment]
A lot of issues were fixed by Serhiy in #106858. Are there more actionable items left? (I did not check the `_decimal` PR yet.)

**Comment by vstinner** [comment]
I suggest to close the issue which has a long history, and maybe create a new issue with a better described scope (list C extensions) if more C extensions need to be fixed.
